pkg/pillar/zedcloud/deferred.go

```diff
@@ -66,55 +85,90 @@ func HandleDeferred(zedcloudCtx *ZedCloudContext, event time.Time, spacing time.
 
 func (ctx *DeferredContext) handleDeferred(log *base.LogObject, event time.Time,
 	spacing time.Duration) bool {
+	ctx.lock.Lock()
+	defer ctx.lock.Unlock()
 
-	log.Functionf("HandleDeferred(%v, %v) map %d\n",
+	if len(ctx.deferredItems) == 0 {
+		return true
+	}
+
+	log.Functionf("HandleDeferred(%v, %v) items %d\n",
 		event, spacing, len(ctx.deferredItems))
-	iteration := 0 // Do some load spreading
-	for key, l := range ctx.deferredItems {
-		log.Functionf("Trying to send for %s items %d\n", key, len(l.list))
-		failed := false
-		for i, item := range l.list {
-			if item.buf == nil {
-				continue
-			}
-			if item.buf.Len() == 0 {
-				log.Errorf("Zero length defered item for %s",
-					key)
-				continue
+
+	//sort items by added time in ascending order
```

Comment:
FWIW "by added time" and "by time the entry was added/created" aka "by time added" results in different interpretation of what "added" means.

Reply:
If the entries retain their createTime (see comments below) and stay at the same index (which the code does in setDeferred), then they are already sorted in the order of the first setDeferred for the object. Hence no need to sort here.

Reply:
Well, I assumed this as time of add/modify, so, we still need to sort them to keep order in case of modifications.